### PR TITLE
(/components/clerk-provider) clarify how ClerkProvider can be used

### PR DESCRIPTION
**Pull request by victoriaxyz**
**Previews**
- https://clerk.com/docs/pr/1759/components/clerk-provider
- https://clerk.com/docs/pr/1759/quickstarts/tanstack-start
- https://clerk.com/docs/pr/1759/quickstarts/react
- https://clerk.com/docs/pr/1759/quickstarts/expo
- https://clerk.com/docs/pr/1759/quickstarts/nextjs
- https://clerk.com/docs/pr/1759/quickstarts/chrome-extension
- https://clerk.com/docs/pr/1759/integrations/databases/convex
- https://clerk.com/docs/pr/1759/guides/waitlist
- https://clerk.com/docs/pr/1759/guides/authjs-migration

**ClerkProvider docs page**
> The `<ClerkProvider>` component is required to integrate Clerk into your React application, providing session and user context to Clerk's hooks and components.
>
> The recommended approach is to wrap your entire app with `<ClerkProvider>` at the entry point to make authentication globally accessible. If you only need authentication for specific routes or pieces of your application, render `<ClerkProvider>` deeper in the component tree. This allows you to implement Clerk's functionality precisely where required without impacting the rest of your app.

**Individual docs section**
>  The [`<ClerkProvider>`](/docs/components/clerk-provider) component provides session and user context to Clerk's hooks and components. It's recommended to wrap your entire app at the entry point with `<ClerkProvider>` to make authentication globally accessible. See the [reference doc](/docs/components/clerk-provider) for other configuration options.

**Updated files**
- [x] docs/components/clerk-provider.mdx
- [x] docs/quickstarts/tanstack-start.mdx
- [x] docs/quickstarts/react.mdx
- [x] docs/quickstarts/expo.mdx
- [x] docs/quickstarts/nextjs.mdx
- [x] docs/quickstarts/chrome-extension.mdx
- [x] docs/integrations/databases/convex.mdx
- [x] docs/guides/waitlist.mdx
- [x] docs/guides/authjs-migration.mdx

### CODE REVIEW

File: docs/components/clerk-provider.mdx [hunk 1]
```diff
@@ -1,47 +1,35 @@
...
 ---
 
-The `<ClerkProvider>` component wraps your React application to provide active session and user context to Clerk's hooks and other components.
+The `<ClerkProvider>` component is required to integrate Clerk into your React application, providing session and user context to Clerk's hooks and components.
+
+The recommended approach is to wrap your entire app with `<ClerkProvider>` at the entry point to make authentication globally accessible. If you only need authentication for specific routes or pieces of your application, render `<ClerkProvider>` deeper in the component tree. This allows you to implement Clerk's functionality precisely where required without impacting the rest of your app.
 
 ## Usage
 
 The `<ClerkProvider>` component must be added to your React entrypoint.
```
Comment (alexisintech) [inline docs/components/clerk-provider.mdx:12]: this isn't exactly accurate, so i'd remove this.

```suggestion
```

File: docs/components/clerk-provider.mdx [hunk 2]
```diff
@@ -1,47 +1,35 @@
...
 
 <Tabs items={["Next.js", "React"]}>
   <Tab>
-    <CodeBlockTabs options={["App Router", "Pages Router"]}>
```
Comment (alexisintech) [inline docs/components/clerk-provider.mdx:14]: Our rule of thumb for Pages Router examples:
Remove unnecessary Pages Router examples for Next.js example. An example is considered unnecessary if it has the same exact code as an App Router example. In contrast, a specific example of being necessary is when an App Router example uses auth() and an Pages Router example uses getAuth(). These can be argued for, as the two different router types require two different methods for retrieving auth information.

for this case, the code is not exactly the same - the layout for an app router, and the `_app.tsx` file for pages router are pretty different. I would keep both.

File: docs/components/clerk-provider.mdx [hunk 3]
```diff
@@ -1,47 +1,35 @@
...
+    ```tsx {{ filename: 'app/layout.tsx' }}
+    import React from 'react'
+    import { ClerkProvider } from '@clerk/nextjs'
+
+    export default function RootLayout({ children }: { children: React.ReactNode }) {
+      return (
+        <html lang="en">
+          <head>
+            <title>Next.js 13 with Clerk</title>
+          </head>
```
Comment (alexisintech) [inline docs/components/clerk-provider.mdx:25]: is there a reason you added this? we don't have this anywhere else, certainly not the quickstart.

File: docs/components/clerk-provider.mdx [hunk 4]
```diff
@@ -1,47 +1,35 @@
...
-
-      export default function RootLayout({ children }: { children: React.ReactNode }) {
-        return (
+    ```tsx {{ filename: 'app/layout.tsx' }}
+    import React from 'react'
+    import { ClerkProvider } from '@clerk/nextjs'
+
+    export default function RootLayout({ children }: { children: React.ReactNode }) {
+      return (
+        <html lang="en">
```
Comment (alexisintech) [inline docs/components/clerk-provider.mdx:22]: we wrap the `html` with the clerkprovider as well - I would look at the quickstart, and the other instances in the docs, for how the clerkprovider is mounted.. in order to maintain consistency..

File: docs/guides/authjs-migration.mdx [hunk 5]
```diff
@@ -69,7 +69,9 @@ This guide shows how to migrate an application using Auth.js (formerly NextAuth.
 
   ### Wrap your Next.js app in `<ClerkProvider />`
 
-  You will need to remove the `<SessionProvider session={session}>` provider from Auth.js and replace it with the `<ClerkProvider />` provider from Clerk. `<ClerkProvider />` will need to wrap your application, or wrap the portion of your app where you want to handle authentication.
+  Remove the `<SessionProvider session={session}>` provider from Auth.js and replace it with `<ClerkProvider />`.
+
+  The [`<ClerkProvider>`](/docs/components/clerk-provider) component provides session and user context to Clerk's hooks and components. It's recommended to wrap your entire app at the entry point with `<ClerkProvider>` to make authentication globally accessible. See the [ClerkProvider docs](/docs/components/clerk-provider) for other configuration options.
```
Comment (alexisintech) [inline docs/guides/authjs-migration.mdx:74]: `<ClerkProvider>` should always be formatted as such.  Just "ClerkProvider" is not accurate, and not used anywhere in the docs. Any piece of code in the docs should **always** be formatted correctly.

But I prefer not to put the docs' title in the link

```suggestion
  The [`<ClerkProvider>`](/docs/components/clerk-provider) component provides session and user context to Clerk's hooks and components. It's recommended to wrap your entire app at the entry point with `<ClerkProvider>` to make authentication globally accessible. See the [reference doc](/docs/components/clerk-provider) for other configuration options.
```

File: docs/guides/waitlist.mdx [hunk 6]
```diff
@@ -80,7 +80,9 @@ In [**Waitlist** mode](/docs/authentication/configuration/restrictions#waitlist)
 
   ### Add `<ClerkProvider>` to your app
 
-  The [`<ClerkProvider>`](/docs/components/clerk-provider) component wraps your app to provide active session and user context to Clerk's hooks and other components. To use the `<Waitlist />` component in your app, you must provide the `waitlistUrl` prop, which points to the URL of your waitlist page.
+  The [`<ClerkProvider>`](/docs/components/clerk-provider) component provides session and user context to Clerk's hooks and components. It's recommended to wrap your entire app at the entry point with `<ClerkProvider>` to make authentication globally accessible. See the [ClerkProvider docs](/docs/components/clerk-provider) for other configuration options.
```
Comment (alexisintech) [inline docs/guides/waitlist.mdx:83]: same comment as above

File: docs/integrations/databases/convex.mdx [hunk 7]
```diff
@@ -104,9 +104,9 @@ This tutorial assumes that you have already [set up a Clerk application](/docs/q
 
   ### Configure the Clerk and Convex providers
 
-  Both Clerk and Convex have Provider components that are required to wrap your React application to provide the authentication and client context.
+  Both Clerk and Convex have Provider components that are required to wrap your React app to provide session and user context to Clerk's hooks and components. It's recommended to wrap your entire app at the entry point with `<ClerkProvider>` to make authentication globally accessible. See the [ClerkProvider docs](/docs/components/clerk-provider) for other configuration options.
```
Comment (alexisintech) [inline docs/integrations/databases/convex.mdx:107]:
```suggestion
  Both Clerk and Convex have provider components that are required to wrap your React app to provide authentication and client context. 
  
  Clerk's provider component is `<ClerkProvider>`, which should wrap your entire app at the entry point to make authentication globally accessible. See the [reference docs](/docs/components/clerk-provider) for other configuration options.
  
  Convex offers a provider that is specifically for integrating with Clerk called [`<ConvexProviderWithClerk>`](https://docs.convex.dev/api/modules/react#convexprovider) .
```

"Both Clerk and Convex have Provider components that are required to wrap your React app to provide session and user context to Clerk's hooks and components." is not accurate. Convex's provider component does not provide session and user context to Clerk's hooks and components - Clerk's provider does. Convex's provider provides Convex auth information.

We can also add some reference links to Convex's docs.

File: docs/integrations/databases/convex.mdx [hunk 8]
```diff
@@ -104,9 +104,9 @@ This tutorial assumes that you have already [set up a Clerk application](/docs/q
...
 
-  You may have already had a `<ConvexProvider>` configured. Be sure that `<ClerkProvider>` wraps `ConvexProviderWithClerk`, and that `useAuth` is passed to `ConvexProviderWithClerk`.
+  You might have already configured `<ConvexProvider>`. Ensure that `<ClerkProvider>` wraps `ConvexProviderWithClerk` and that `useAuth` is passed to `ConvexProviderWithClerk`.
```
Comment (alexisintech) [inline docs/integrations/databases/convex.mdx:109]:
```suggestion
  The following example demonstrates how to configure Clerk and Convex's providers. Clerk's `useAuth()` hook must be passed to `<ConvexProviderWithClerk>` and Clerk's `<ClerkProvider>` must be wrapped around it.
```

File: docs/quickstarts/chrome-extension.mdx [hunk 9]
```diff
@@ -79,7 +79,7 @@ description: Add authentication and user management to your Chrome Extension wit
 
   ### Add `<ClerkProvider>` to your app
 
-  All Clerk hooks and components must be children of the [`<ClerkProvider>`](/docs/components/clerk-provider) component, which provides active session and user context. Wrap your app in the `<ClerkProvider>` component and pass your Publishable Key as a prop.
+  The [`<ClerkProvider>`](/docs/components/clerk-provider) component provides session and user context to Clerk's hooks and components. It's recommended to wrap your entire app at the entry point with `<ClerkProvider>` to make authentication globally accessible. See the [ClerkProvider docs](/docs/components/clerk-provider) for other configuration options.
```
Comment (alexisintech) [inline docs/quickstarts/chrome-extension.mdx:82]: same comment as before - please apply to all quickstarts.

File: docs/guides/authjs-migration.mdx [hunk 10]
```diff
@@ -69,7 +69,9 @@ This guide shows how to migrate an application using Auth.js (formerly NextAuth.
 
   ### Wrap your Next.js app in `<ClerkProvider />`
 
-  You will need to remove the `<SessionProvider session={session}>` provider from Auth.js and replace it with the `<ClerkProvider />` provider from Clerk. `<ClerkProvider />` will need to wrap your application, or wrap the portion of your app where you want to handle authentication.
+  Remove the `<SessionProvider session={session}>` provider from Auth.js and replace it with `<ClerkProvider />`.
+
+  The [`<ClerkProvider>`](/docs/components/clerk-provider) component provides session and user context to Clerk's hooks and components. It's recommended to wrap your entire app at the entry point with `<ClerkProvider>` to make authentication globally accessible. See the [reference doc](/docs/components/clerk-provider) for other configuration options.
```
Comment (alexisintech) [inline docs/guides/authjs-migration.mdx:74]:
```suggestion
<Include src="_partials/clerk-provider" />
```

**Review by alexisintech — APPROVED** [review]

**Review by mwickett — APPROVED** [review]
🎸

**Pull request merged**